src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -196,6 +197,7 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
             var discoverable = this.peerAddresses.Values.Where(p => p.LastDiscoveredFrom < this.dateTimeProvider.GetUtcNow().AddHours(-PeerSelector.DiscoveryThresholdHours));
+
```

Comment:
This method does not filter banned peers?

Reply:
@majikandy Didn't you fix this? We discussed this the other day.. unless its done somewhere else now?

Reply:
@majikandy Oh no wait that's another PR, sorry.

Reply:
I can fix SelectedPeersForDiscovery as follows :- (this is on my branch)

![image](https://user-images.githubusercontent.com/16000683/37713732-b9d9a0c8-2d0f-11e8-9b7d-24448623dcdf.png)

Reply:
>@majikandy Didn't you fix this? We discussed this the other day.. unless its done somewhere else now?

I added the filter for !IsSelf in this PR which is not merged yet https://github.com/stratisproject/StratisBitcoinFullNode/pull/1218